File: 0141: linkedListCycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        
        set<ListNode*> check;
        ListNode* current = head;
        while(current != nullptr){

            if(check.count(current)){

                return true;
            }

            check.insert(current);
            current = current->next;
        }

        return false;
    }
};

File: 0141: linkedListCycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        pointers = set()
        curr = head
        while curr:
            if curr in pointers:
                return True
            else:
                pointers.add(curr)
                curr = curr.next
        return False